pkg/controllers/termination/terminate.go

```diff
@@ -77,7 +81,15 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
 		}
 		// Don't attempt to evict a pod that's already evicting
 		if !pod.DeletionTimestamp.IsZero() {
-			continue
+			// If the kubelet cannot register a pod as deleted, we need to force delete the pod in order to terminate the node
+			if time.Since(pod.DeletionTimestamp.Time) > time.Duration(*pod.DeletionGracePeriodSeconds) {
```

Comment:
Duration defaults to nanoseconds, so this comparison is going to be super wrong IIUC.

If DeletionTimestamp is zero, this will never be greater than the deletiongraceperiodseconds (even if they're both zero). Can we collapse this into a single check like:

```
if time.Since(pod.DeletionTimestamp.Time) > time.Duration(ptr.Int64Value(pod.DeletionGracePeriodSeconds) {
   // kubectl delete
}
```

Reply:
Good callout on the default for nanoseconds. 

This line: `if !pod.DeletionTimestamp.IsZero() {` refers to a pod evicting. It's possible for it to be nonZero (meaning it's evicting), but for it to not be past the podDeletionGracePeriodSeconds yet, so we shouldn't collapse this into one line.